### PR TITLE
Issue/8497

### LINKED ISSUE #8290: Remove Jilt integration callouts on Emails setting
### Explain your enhancement (please be detailed)
In #7467 we added Jilt to have its own section in the Downloads > Settings > Emails settings.

We need to remove this call out in the 3.0 release.

### LINKED ISSUE #8497: Add Recapture promo & plugin background install
### Explain your enhancement (please be detailed)

In **Settings > Email > General**, we need to add copy and a "connect" button for our new Recapture partnership. It should be our 3rd option on the page, replacing Jilt.

This needs to go in the next 2.10.x release of EDD if at all possible.

Related issues that probably make sense to do at the same time:
- #8290 
- #8498

Subheading:
**Recapture**

Label:
**Abandoned cart recovery**

Text:
We recommend Recapture for recovering lost revenue by automatically sending effective, targeted emails to customers who abandon their shopping cart. [Learn more here](https://recapture.io/abandoned-carts-easy-digital-downloads) (free trial available).

**[Connect with Recapture]** <-- call-to-action button

---
- The CTA button background installs the Recapture for EDD plugin, then redirects to https://recapture.io/register. This mimics the SendWP (and soon deprecated Jilt) behavior on the same settings page.

- Once the Recapture for EDD plugin is activated, instead of a CTA button, show a “notice-success” styled message.

> Recapture for EDD plugin activated. [Access your Recapture account.](https://recapture.io/account)

This mimics the way we replace the connect button with a notice for the Stripe connection workflow.
image

- The external links (i.e. "Learn more here" and "Access your Recapture account.") should open in a new tab.

Final order on email settings page should be:
- Email Configuration
- SendWP
- Recapture

Current page layout:
image

### LINKED ISSUE #8498: Update SendWP promo
### Explain your enhancement (please be detailed)

In **Settings > Email > General**, we need to:

- Move the SendWP promo to 2nd on the list (between Email Configuration and soon Recapture).
- Update copy in subheading, label, main text and CTA button.
- Display a notice in place of button after SendWP plugin is activated.

This needs to go in the next 2.10.x release of EDD if at all possible.

Related issues that probably make sense to do at the same time:
- #8290 
- #8497

Subheading:
**SendWP** <-- remove "Settings"

Label:
**Deliverability settings**

Text:
We recommend SendWP to ensure quick and reliable delivery of all emails sent from your store, such as purchase receipts, subscription renewal reminders, password resets, and more. [Learn more here](https://sendwp.com/) (free trial available).

**[Connect with SendWP]** <-- add "with"

---
The CTA button background _already_ installs the SendWP plugin and redirects to https://sendwp.com/_/signup/register (no change).

To add: Once the SendWP plugin is activated, instead of a CTA button, show a “notice-success” styled message:

> SendWP plugin activated. [Access your SendWP account.](https://sendwp.com/account/)

- The external links (i.e. "Learn more here" and "Access your SendWP account.") should open in a new tab.

This mimics the way we replace the connect button with a notice for the Stripe connection workflow.
image

Final order on email settings page should be:
- Email Configuration
- SendWP
- Recapture

Current page layout:
image

### PR DESCRIPTION
**Pull request by cklosowski**
Fixes #8497 
Fixes #8290 
Fixes #8498 

Proposed Changes:
1. Moves the SendWP integration callouts to the bottom of the Email General Settnigs
2. Adjusts the copy for the SendWP integration prior to connectnig
3. Adds a styling around the successful connection of SendWP
4. Removes the Jilt callout in the advanced email settings
5. Adds deprecation notices for the Jilt callouts
6. Adds the Recapture callout in the Advanced Email settings

### CODE REVIEW

File: includes/admin/settings/register-settings.php [hunk 1]
```diff
@@ -2205,65 +2216,63 @@ function edd_sendwp_callback($args) {
...
+	ob_start();
 
-		<?php if ( $connected ) : ?>
+	echo $args['desc'];
 
+	// Output the appropriate button and label based on connection status
+	if ( $client_connected ) :
+		$connection_comlpete = get_option('recapture_api_key');
+		?>
+		<div class="inline notice notice-<?php echo $connection_comlpete ? 'success' : 'warning'; ?>">
```
Comment (ashleyfae) [inline includes/admin/settings/register-settings.php:2242]:
```suggestion
		<div class="inline notice notice-<?php echo $connection_complete ? 'success' : 'warning'; ?>">
```

File: includes/admin/settings/register-settings.php [hunk 2]
```diff
@@ -2205,65 +2216,63 @@ function edd_sendwp_callback($args) {
...
 			<p>
-				<button id="edd-jilt-disconnect" class="button"><?php esc_html_e( 'Disconnect Jilt', 'easy-digital-downloads' ); ?></button>
+				<?php _e( 'Recapture plugin activated.', 'easy-digital-downloads' ); ?>
+				<?php printf( __( '%sAccess your Recapture account%s.', 'easy-digital-downloads' ), '<a href="https://recapture.io/account" target="_blank" rel="noopener noreferrer">', '</a>' ); ?>
 			</p>
 
+			<?php if ( $connection_comlpete ) : ?>
```
Comment (ashleyfae) [inline includes/admin/settings/register-settings.php:2248]:
```suggestion
			<?php if ( $connection_complete ) : ?>
```

File: includes/admin/settings/register-settings.php [hunk 3]
```diff
@@ -2205,65 +2216,63 @@ function edd_sendwp_callback($args) {
...
+		<p>
+			<?php _e( 'We recommend Recapture for recovering lost revenue by automatically sending effective, targeted emails to customers who abandon their shopping cart.', 'easy-digital-downloads' ); ?>
+		</p>
+		<p>
+			<?php printf( __( '%sLearn More%s (Free trial available)' ), '<a href="https://recapture.io/abandoned-carts-easy-digital-downloads" taregt="_blank" rel="noopener noreferrer">', '</a>' ); ?>
+		</p>
 		<p>
-			<button id="edd-jilt-connect" class="button button-primary">
-				<?php esc_html_e( 'Install Jilt', 'easy-digital-downloads' ); ?>
+			<button type="button" id="edd-recapture-connect" class="button button-primary"><?php esc_html_e( 'Connect with Recapture', 'easy-digital-downloads' ); ?>
```
Comment (ashleyfae) [inline includes/admin/settings/register-settings.php:2268]: We should only show this button if `current_user_can( 'install_plugins' )`. IMO we could probably show the above "We recommend Recapture..." text, just shouldn't output the button.

**Review by ashleyfae — CHANGES_REQUESTED** [review]
There are a few other Jilt functions in `includes/emails/functions.php` that we probably want to deprecate:

- `edd_jilt_remote_install_handler()`
- `edd_jilt_connect_handler()`
- `edd_jilt_disconnect_handler()`
- `maybe_add_jilt_notice_to_abandoned_payment()` (do we want an equivalent for Recapture here?)

File: includes/admin/settings/register-settings.php [hunk 4]
```diff
@@ -2205,65 +2216,63 @@ function edd_sendwp_callback($args) {
...
 			</p>
-
-		<?php else : ?>
-
+			<?php else : ?>
 			<p>
-				<a id="edd-jilt-connect" class="button button-primary" href="<?php echo esc_url( $connect_url ); ?>">
-					<?php esc_html_e( 'Connect to Jilt', 'easy-digital-downloads' ); ?>
-				</a>
+				<?php printf( __( 'Complete your connection to Recapture %shere%s.', 'easy-digital-downloads' ), '<a href="' . admin_url( 'admin.php?page=recapture' ) . '">', '</a>' ); ?>
```
[the remark this answers is not in the document]
Comment (robincornett) [reply includes/admin/settings/register-settings.php:2254]: Yes, "here" is not accessible link text.

[the remark this answers is not in the document]
Comment (cklosowski) [reply includes/admin/settings/register-settings.php:2254]: I'm matching what Recapture uses for their terminology here, trying to stay within the way they document their users through the connection process.

File: includes/admin/settings/register-settings.php (hunk 3 above)
Comment (cklosowski) [reply includes/admin/settings/register-settings.php:2268]: :+1:

**Comment by cklosowski** [comment]
> There are a few other Jilt functions in `includes/emails/functions.php` that we probably want to deprecate:
> 
>     * `edd_jilt_remote_install_handler()`
> 
>     * `edd_jilt_connect_handler()`
> 
>     * `edd_jilt_disconnect_handler()`
> 
>     * `maybe_add_jilt_notice_to_abandoned_payment()` (do we want an equivalent for Recapture here?)

yeah I'll get those moved to deprecated as well. It was not part of the agreement to have a Recapture equivalent added into this location. I'm just going off what was requested by the team.

**Comment by cklosowski** [comment]
@ashleyfae Added on the abandoned payments:
image

It is able to be dismissed.

File: includes/admin/settings/register-settings.php [hunk 5]
```diff
@@ -2205,65 +2216,65 @@ function edd_sendwp_callback($args) {
 }
 
 /**
- * Jilt Callback
+ * Recaptuer Callback
```
Comment (ashleyfae) [inline includes/admin/settings/register-settings.php:2219]:
```suggestion
 * Recapture Callback
```

File: includes/deprecated-functions.php [hunk 6]
```diff
@@ -731,3 +731,265 @@ function edd_get_paypal_page_style() {
...
+			wp_kses_post(
+					sprintf(
+					/* Translators: %1$s - <a> tag, %2$s - </a> tag */
+							__( '%1$sClick here%2$s to visit your Jilt dashboard', 'easy-digital-downloads' ),
+							'<a href="' . esc_url( $account_url ) . '" target="_blank">',
+							'</a>'
+					)
+			);
+			?>
+		</p>
```
Comment (ashleyfae) [inline includes/deprecated-functions.php:776]: The above code looks double indented.

File: includes/deprecated-functions.php [hunk 7]
```diff
@@ -731,3 +731,265 @@ function edd_get_paypal_page_style() {
...
+							'<span class="screen-reader-text">',
+							'</span>
+					</a>'
+					)
+			);
+			?>
+		</div>
+		<?php
+	}
+}
```
Comment (ashleyfae) [inline includes/deprecated-functions.php:995]: Lots more double indenting above, I think.

File: includes/admin/settings/register-settings.php [hunk 8]
```diff
@@ -2183,18 +2188,24 @@ function edd_sendwp_callback($args) {
...
 		<?php
 	else :
 		?>
-
 		<p>
-			<button type="button" id="edd-sendwp-connect" class="button button-primary"><?php esc_html_e( 'Connect SendWP', 'easy-digital-downloads' ); ?>
+			<?php _e( 'We recommend SendWP to ensure quick and reliable delivery of all emails sent from your store, such as purchase receipts, subscription renewal reminders, password resets, and more.', 'easy-digital-downloads' ); ?>
+		</p>
+		<p>
+			<?php printf( __( '%sLearn More%s (Free trial available)', 'easy-digital-downloads' ), '<a href="https://sendwp.com/" taregt="_blank" rel="noopener noreferrer">', '</a>' ); ?>
```
Comment (ashleyfae) [inline includes/admin/settings/register-settings.php:2205]:
```suggestion
			<?php printf( __( '%sLearn More%s (Free trial available)', 'easy-digital-downloads' ), '<a href="https://sendwp.com/" target="_blank" rel="noopener noreferrer">', '</a>' ); ?>
```

File: includes/admin/settings/register-settings.php [hunk 9]
```diff
@@ -2205,65 +2216,65 @@ function edd_sendwp_callback($args) {
...
+	<?php
+	else :
+		?>
+		<p>
+			<?php _e( 'We recommend Recapture for recovering lost revenue by automatically sending effective, targeted emails to customers who abandon their shopping cart.', 'easy-digital-downloads' ); ?>
+		</p>
 		<p>
-			<button id="edd-jilt-connect" class="button button-primary">
-				<?php esc_html_e( 'Install Jilt', 'easy-digital-downloads' ); ?>
+			<?php printf( __( '%sLearn More%s (Free trial available)', 'easy-digital-downloads' ), '<a href="https://recapture.io/abandoned-carts-easy-digital-downloads" taregt="_blank" rel="noopener noreferrer">', '</a>' ); ?>
```
Comment (ashleyfae) [inline includes/admin/settings/register-settings.php:2265]:
```suggestion
			<?php printf( __( '%sLearn More%s (Free trial available)', 'easy-digital-downloads' ), '<a href="https://recapture.io/abandoned-carts-easy-digital-downloads" target="_blank" rel="noopener noreferrer">', '</a>' ); ?>
```

File: includes/admin/settings/register-settings.php (hunk 8 above)
Comment (ashleyfae) [reply includes/admin/settings/register-settings.php:2205]: Also, I don't see a free trial available? According to their website you can:

> Try it for 14 days for only $1.00

So it's more like a paid trial.

**Comment by ashleyfae** [comment]
Left a few more notes, but they're all minor so I started testing functionality:

## Tested

- [x] Connecting to SendWP
- [x] Connecting to Recapture
    - [x] Notice after Recapture is installed, but before connected.
    - [x] Notice after Recapture is fully connected.
    - [x] Disconnect.
    - [x] Abandoned payment notice when Recapture is not connected.
    - [x] Abandoned payment notice does not appear when Recapture is connected.

**Comment by pderksen** [comment]
Some additional tweaks that shouldn't affect testing.

image

- "Advanced Emails" section heading should be removed.
- SendWP's "[Learn more](url)."
  - Keep on same line as previous text (no line break).
  - Lowercase "m" in more.
  - Link opens in new tab.
  - Remove "(free trial available)". Great catch @ashleyfae!
- Recapture's "[Learn more](url) (free trial available)."
  - Keep on same line as previous text (no line break).
  - Lowercase "m" in more and "f" in free.
  - Link opens in new tab.
- `Connect with Recapture` button should be same larger size button as `Connect with SendWP`

**Comment by cklosowski** [comment]
@pderksen Looks like your CSS is cached,hard refresh and the button size should be correct, or have the script debug enabled, as we haven't minified the CSS yet.
image

File: includes/admin/settings/register-settings.php (hunk 8 above)
Comment (cklosowski) [reply includes/admin/settings/register-settings.php:2205]: image

image

**Comment by cklosowski** [comment]
@pderksen here are the changes:
image

File: includes/admin/settings/register-settings.php [hunk 10]
```diff
@@ -2199,10 +2194,7 @@ function edd_sendwp_callback($args) {
 	else :
 		?>
 		<p>
-			<?php _e( 'We recommend SendWP to ensure quick and reliable delivery of all emails sent from your store, such as purchase receipts, subscription renewal reminders, password resets, and more.', 'easy-digital-downloads' ); ?>
-		</p>
-		<p>
-			<?php printf( __( '%sLearn More%s (Free trial available)', 'easy-digital-downloads' ), '<a href="https://sendwp.com/" taregt="_blank" rel="noopener noreferrer">', '</a>' ); ?>
+			<?php _e( 'We recommend SendWP to ensure quick and reliable delivery of all emails sent from your store, such as purchase receipts, subscription renewal reminders, password resets, and more.', 'easy-digital-downloads' ); ?> <?php printf( __( '%sLearn more%s (Free trial available)', 'easy-digital-downloads' ), '<a href="https://sendwp.com/" target="_blank" rel="noopener noreferrer">', '</a>' ); ?>
```
Comment (ashleyfae) [inline includes/admin/settings/register-settings.php:2197]:
```suggestion
			<?php _e( 'We recommend SendWP to ensure quick and reliable delivery of all emails sent from your store, such as purchase receipts, subscription renewal reminders, password resets, and more.', 'easy-digital-downloads' ); ?> <?php printf( __( '%sLearn more%s', 'easy-digital-downloads' ), '<a href="https://sendwp.com/" target="_blank" rel="noopener noreferrer">', '</a>' ); ?>
```

This one doesn't have a free trial.

**Review by ashleyfae — CHANGES_REQUESTED** [review]

File: includes/admin/settings/register-settings.php (hunk 8 above)
Comment (pderksen) [reply includes/admin/settings/register-settings.php]: To be clear, Recapture has a free trial, but SendWP does not. I was incorrect earlier.

**Comment by pderksen** [comment]
> @ashleyfae Added on the abandoned payments:
> image
> 
> It is able to be dismissed.

For clarify, this appears in individual payment details with the abandoned status, and links to https://recapture.io/abandoned-carts-easy-digital-downloads in a new tab.
image

**Comment by cklosowski** [comment]
Thanks for removing that free trial available @ashleyfae I think this is ready for final review.

**Review by ashleyfae — APPROVED** [review]
In the interest of time, I pushed up a couple more double indentation fixes. PhpStorm _really_ wanted to double indent those for some reason.

**Comment by cklosowski** [comment]
@ashleyfae thanks, good to know it wasn't just mine PHPstorm doing that.

**Pull request merged**